src/extension/background-script/actions/allowances/update.ts

```diff
@@ -1,12 +1,18 @@
+import type { MessageAllowanceUpdate } from "~/types";
+
 import db from "../../db";
 
-const updateAllowance = async (message, sender) => {
+const updateAllowance = async (message: MessageAllowanceUpdate) => {
   const id = parseInt(message.args.id);
 
-  const update = {};
+  const update: {
+    totalBudget?: number;
+    remainingBudget?: number;
+    enabled?: boolean;
+  } = {};
   if (Object.prototype.hasOwnProperty.call(message.args, "totalBudget")) {
-    update.totalBudget = parseInt(message.args.totalBudget);
-    update.remainingBudget = parseInt(message.args.totalBudget);
+    update.totalBudget = message.args.totalBudget;
+    update.remainingBudget = message.args.totalBudget;
```

Comment:
is it reliable that those are numbers now? 
because those functions are called through the extension messaging system (e.g. `utils.call()`) I think TypeScript can not verify that, can it?